src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -1132,20 +1132,27 @@ class query extends viewModelBase {
         this.graphQueryResults.onResize();
     }
     
-    exportCsv() {
+
+    exportCsvVisibleColumns() {
+        const columns = this.columnsSelector.getSimpleColumnsFields();
+        this.exportCsvInternal(columns);
+    }
+
+    exportCsvFull() {
+        this.exportCsvInternal();
+    }
+
+    private exportCsvInternal(columns?: string[]) {
         eventsCollector.default.reportEvent("query", "export-csv");
 
-        let args: { format: string, debug?: string };
+        let args: { format: string, debug?: string, field?: string[] };
+
         if (this.criteria().indexEntries()) {
-            args = {
-                format: "csv",
-                debug: "entries"
-            };
+            args = columns ? { format: "csv", debug: "entries",field: columns } : { format: "csv", debug: "entries"};
```

Comment:
args = { format: "csv", debug: "entries",field: columns }

if columsn will be undefined it won't be send to server anyway

Reply:
done